pkg/node/manager/manager.go

```diff
@@ -49,6 +48,19 @@ type nodeEntry struct {
 	node  node.Node
 }
 
+// IPCache is the set of interactions the node manager performs with the ipcache
+type IPCache interface {
+	Upsert(ip string, hostIP net.IP, hostKey uint8, k8sMeta *ipcache.K8sMetadata, newIdentity ipcache.Identity) bool
+	Delete(IP string, source source.Source)
+}
+
+// Configugration is the set of configuration options the node manager depends
+// on
+type Configugration interface {
```

Comment:
Any special reason why this is an interface and not a structure? A structure allows to add more fields to be used in the future and does not require `option.Config` to implement every option the Manager might use.

Reply:
Yes, the reason is to not assume that options are immutable forever.